test/e2e/lifecycle/node_auto_repairs.go

```diff
@@ -0,0 +1,166 @@
+		By("Wait till node is unready.")
+		Expect(framework.WaitForNodeToBeNotReady(c, nodeName, defaultTimeout)).To(Equal(true))
+		By("Wait till node is repaired.")
+		Expect(framework.WaitForNodeToBeReady(c, nodeName, repairTimeout)).To(Equal(true))
+	})
+})
+
+func execCmd(args ...string) *exec.Cmd {
+	glog.Infof("Executing: %s", strings.Join(args, " "))
+	return exec.Command(args[0], args[1:]...)
+}
+
+func getReadyNodes(c clientset.Interface) []v1.Node {
+	nodeList := framework.GetReadySchedulableNodesOrDie(c)
+	return nodeList.Items
+}
+
+func enableAutoRepair(nodePool string) error {
+	glog.Infof("Using gcloud to enable auto repair for pool %s", nodePool)
+	output, err := execCmd("gcloud", "beta", "container", "node-pools", "update", nodePool,
+		"--enable-autorepair",
+		"--project="+framework.TestContext.CloudConfig.ProjectID,
+		"--zone="+framework.TestContext.CloudConfig.Zone,
+		"--cluster="+framework.TestContext.CloudConfig.Cluster).CombinedOutput()
+	if err != nil {
+		glog.Errorf("Failed to enable auto repair: %s", string(output))
+		return fmt.Errorf("failed to enable auto repair: %v", err)
+	}
+	return nil
+}
+
+func disableAutoRepair(nodePool string) error {
+	glog.Infof("Using gcloud to disable auto repair for pool %s", nodePool)
+	output, err := execCmd("gcloud", "beta", "container", "node-pools", "update", nodePool,
+		"--no-enable-autorepair",
+		"--project="+framework.TestContext.CloudConfig.ProjectID,
+		"--zone="+framework.TestContext.CloudConfig.Zone,
+		"--cluster="+framework.TestContext.CloudConfig.Cluster).CombinedOutput()
+	if err != nil {
+		glog.Errorf("Failed to disable auto repair: %s", string(output))
+		return fmt.Errorf("failed to disable auto repair: %v", err)
+	}
+	return nil
+}
+
+func stopKubeletOnNode(node string) error {
+	glog.Infof("Using gcloud to stop Kublet on node %s", node)
+	output, err := execCmd("gcloud", "compute", "ssh", node,
+		"--command=sudo systemctl stop kubelet-monitor.service && sudo systemctl stop kubelet.service",
+		"--zone="+framework.TestContext.CloudConfig.Zone).CombinedOutput()
+	if err != nil {
+		glog.Errorf("Failed to stop Kubelet: %v", string(output))
+		return fmt.Errorf("failed to stop Kubelet: %v", err)
+	}
+	return nil
+}
+
+func recreateNode(nodeName string, groupName string) error {
+	glog.Infof("Using gcloud to recreate node %s", nodeName)
+	//gcloud compute instance-groups managed recreate-instances gke-oneoff-default-pool-e4383993-grp --instances=gke-oneoff-default-pool-e4383993-81jm --zone=us-central1-c
+	output, err := execCmd("gcloud", "compute", "instance-groups", "managed", "recreate-instances", groupName,
+		"--instances="+nodeName,
+		"--zone="+framework.TestContext.CloudConfig.Zone).CombinedOutput()
+	if err != nil {
+		glog.Errorf("Failed to recreate node: %s", string(output))
+		return fmt.Errorf("failed to recreate node: %v", err)
+	}
+	return nil
+}
+
+func isNodeReady(c clientset.Interface, nodeName string) (bool, error) {
```

Comment:
Are you sure we have any library function that does this already?

Reply:
I didn't manage to find it.